Backend/lib/message-service/package.yaml

```diff
@@ -0,0 +1,130 @@
+name: message-service
+version: 0.1.0.0
+github: "nammayatri/nammayatri"
+license: AllRightsReserved
+author: "nammayatri"
+maintainer: "nammayatri"
+copyright: "2023 Juspay Technologies Private Limited"
+
+# Metadata used when publishing your package
+# synopsis:            Short description of your package
+# category:            Web
+
+# To avoid duplicated efforts in documentation and dealing with the
+# complications of embedding Haddock markup inside cabal files, it is
+# common to point users to the README.md file.
+description: Please see the README on GitHub at <https://github.com/nammayatri/nammayatri#readme>
+
+default-extensions:
+  - ConstraintKinds
+  - DataKinds
+  - DefaultSignatures
+  - DeriveAnyClass
+  - DeriveFunctor
+  - DeriveGeneric
+  - DuplicateRecordFields
+  - ExplicitNamespaces
+  - FlexibleContexts
+  - FlexibleInstances
+  - FunctionalDependencies
+  - GADTs
+  - LambdaCase
+  - MultiParamTypeClasses
+  - MultiWayIf
+  - NamedFieldPuns
+  - NoImplicitPrelude
+  - OverloadedStrings
+  - PatternSynonyms
+  - PolyKinds
+  - RankNTypes
+  - RecordWildCards
+  - ScopedTypeVariables
+  - TupleSections
+  - TypeApplications
+  - TypeFamilies
+  - TypeOperators
+  - ViewPatterns
+  - BlockArguments
+  - TypeSynonymInstances
+  - UndecidableInstances
+  - StandaloneDeriving
+  - PackageImports
+
+dependencies:
+  - aeson
+  - base >= 4.7 && < 5
+  - generic-lens
+  - bytestring
+  - cryptonite
+  - containers
+  - euler-hs
+  - exceptions
+  - hspec
+  - http-client
+  - lens
+  - servant
+  - servant-client
+  - servant-client-core
+  - servant-server
+  - servant-multipart
+  - text
+  - wai
+  - warp
+  - time
+  - stm
+  - resource-pool
+  - uuid
+  - prometheus-client
+  - record-hasfield
+  - record-dot-preprocessor
+  - hashmap
+  - servant-openapi3
+  - openapi3
+  - string-conversions
+  - either
+  - persistent
+  - text-conversions
+  - esqueleto
+  - beckn-services
+  - unordered-containers
+  - dashboard-helper-api
+  - shared-services
+  - extra
+  - scheduler
+  - singletons
+  - cassava
+  - vector
+
+ghc-options:
+  - -fwrite-ide-info
+  - -hiedir=.hie
+  - -fplugin=RecordDotPreprocessor
+
+library:
+  source-dirs: src
+  ghc-options:
+    - -Wall
+    - -Wcompat
+    - -Widentities
+    - -fhide-source-paths
+    - -Werror
+  dependencies:
+    - mobility-core
+    - beckn-spec
+    - location-updates
+    - special-zone-a
+
+executables:
+  dynamic-offer-driver-app-exe:
+    main: Main.hs
+    source-dirs: server
+    other-modules: []
+    ghc-options:
+      - -threaded
+      - -rtsopts
+      - '"-with-rtsopts=-N -T"'
+    dependencies:
+      - dynamic-offer-driver-app
```

Comment:
This is a cyclic dependency. `dynamic-offer-driver-app` in turn depends on `message-service`. This explains likely why the Nix build fails.